src/connector/src/source/kinesis/enumerator/client.rs

```diff
@@ -87,8 +87,8 @@ impl SplitEnumerator for KinesisSplitEnumerator {
             .map(|x| KinesisSplit {
                 shard_id: x.shard_id().to_string().into(),
                 // handle start with position in reader part
-                start_position: KinesisOffset::None,
-                end_position: KinesisOffset::None,
+                next_offset: KinesisOffset::None,
+                end_offset: KinesisOffset::None,
```

Comment:
you can see from 
https://github.com/risingwavelabs/risingwave/blob/71732b1c4446d216f500267a9e1f2bf426a88519/src/connector/src/source/kinesis/source/reader.rs#L255-L259

if kinesis read from a position, in current impl, it should begin with the next one. 
So `next_offset` does not mean next offset

Reply:
Hah?

Reply:
`AfterSequenceNumber` is a way to denote the next one after the given sequence number. I don't see any problem here...

Reply:
Try read this in English: The "next offset" is the offset `AfterSequenceNumber("abc")`.

Reply:
I understand you want to eliminate the concept misalignment in the sources' split def, it is good for the project.

The variable name ought to describe the value it stores, not involving how other components do with it. But here, it stores the latest successful read offset (and literally the start offset if begin with Timestamp or TrimHorizen). It is not the next offset. 
Kinesis does have its own primitive to begin consumption with the offset after the given one, your proposal may not take this for granted.

> Try read this in English
Besides this, I know English well and please remain calm and friendly in communications. Be professional, dude.

Reply:
I was not calm because this issue is too obvious and had been discussed for quite a long time.

It's obvious that the split metadata should store `next offset`, which is basically the same meaning as `start offset` when recovery involves. The problem here is that we don't know the next offset after we got a actually offset, so what we can do is to store a `after(offset)` to denote the next offset, and that is what Kinesis's `AFTER_SEQUENCE_NUMBER` mean.

Reply:
> The core change is `start_position = SequenceNumber(abc)` to `start_position = AfterSequenceNumber(abc)`.

This is pretty straightforward as you can see the match arm is changed from:

```rust
KinesisOffset::SequenceNumber(seq) => ShardIteratorType::AfterSequenceNumber,
```

to

```rust
KinesisOffset::AfterSequenceNumber(seq) => ShardIteratorType::AfterSequenceNumber,
```

Reply:
In your logic, if `start_position` is last seen offset, then the `Earliest` and `Latest` variants should be `BeforeEarliest`, `BeforeLatest`. Is it clear enough?

Reply:
> In your logic, if start_position is last seen offset, then the Earliest and Latest variants should be BeforeEarliest, BeforeLatest. Is it clear enough?

Using sarcasm and rhetorical questions in the workplace is not a friendly communication attitude.

> Is it clear enough?

Yes, I fully understand your motivation and proposed changes. 

I fully understand the change and will give an approve

Reply:
> > In your logic, if start_position is last seen offset, then the Earliest and Latest variants should be BeforeEarliest, BeforeLatest. Is it clear enough?
> 
> Using sarcasm and rhetorical questions in the workplace is not a friendly communication attitude.

Hah? I'm not using any sarcasm and rhetorical question. "Is it clear enough?" just means "这么解释足够清楚了吗？".

Reply:
Don't be so sensitive. I'm just trying to explain. I admit that I was a bit impatient at the very beginning of this thread. But I didn't mean to be any aggressive after that.

Reply:
> Yes, I fully understand your motivation and proposed changes.

Ok, then I think we have reached a consensus.